### PR TITLE
feat: support Angular 13

### PR DESCRIPTION
**Pull request by arturovt**
* packages have been updated
* Angular 13 no longer generates UMD bundles, it generates es2015 and es2020 bundles and outputs `.mjs` files **only**
* schematics have been updated to add `style-loader` since it's not a dependency of ngtools/webpack anymore
* scripts have been updated from `--prod` to `--configuration production`, since `--prod` is deprecated
* schematics have been updated to get the `VERSION` correctly (see code comments)

Will require publishing major version for Angular 13 since it's not backwards compatible with older versions.

### CODE REVIEW

**Review by tscislo — APPROVED** [review]

File: libs/single-spa-angular/webpack/index.ts
Comment (jogelin) [inline libs/single-spa-angular/webpack/index.ts:38]: Shouldn't we use the directory options instead? 
https://webpack.js.org/configuration/dev-server/#directory

```javascript
devServer: {
    static: {
      directory: path.resolve(__dirname, "static"),
```

**Comment by jogelin** [comment]
@arturovt I have a question concerning the [shared Angular libraries explained here](https://single-spa.js.org/docs/ecosystem-angular/#shared-angular). Is it still valid with Angular 13?

I mean it is using esbuild and it looks like it doesn't produce `systemjs` bundle yet...

**Comment by arturovt** [comment]
@jogelin I'm honestly not sure. I'll need to re-check that. Considering that VE support is dropped since Angular 13 I **could guess* that it's possible to share dependencies through SystemJS, as I said I have to double check that.

File: libs/single-spa-angular/webpack/index.ts
Comment (arturovt) [reply libs/single-spa-angular/webpack/index.ts:38]: I took that from your comment where you were saying that you patched single-spa config. I don't mind changing that to `static.directory`. Is there a difference between these 2 approaches? I'm asking because I'm unaware of that.

Comment (jogelin) [reply libs/single-spa-angular/webpack/index.ts:38]: > I took that from your comment where you were saying that you patched single-spa config

Indeed but I upgraded again and then Webpack complained again about the schema. I'll update my comment

Comment (jogelin) [reply libs/single-spa-angular/webpack/index.ts:38]: Comment updated with:
```javascript
  const singleSpaWebpackConfig = singleSpaAngularWebpack(config, options);

  const singleSpaWebpackConfigPatched = {
    ...singleSpaWebpackConfig,
    devServer: {
      ...singleSpaWebpackConfig.devServer,
      static: {
          directory: singleSpaWebpackConfigPatched.devServer.contentBase,
        },
      }
    },
  };
  delete singleSpaWebpackConfigPatched.devServer.contentBase;
```

but personally, I don't know why we need that?! I am in Nx repo and it is not even compatible to target the root src/...

**Review by jogelin — APPROVED** [review]

**Review by jogelin — APPROVED** [review]

**Comment by jogelin** [comment]
@joeldenning could you have a look?

**Comment by jogelin** [comment]
@arturovt this PR will also solve the new angular format?
image

**Comment by ilteoood** [comment]
Hi @joeldenning, did you had time to take a look at this?

**Comment by chucknelson** [comment]
> @arturovt this PR will also solve the new angular format?

Just wanted to chime in since @arturovt is probably busy with other things haha - this brings the library in line with what Angular 13 has standardized to:
1. Complies as Ivy now instead of view engine - per the changes in [tsconfig.lib.json](https://github.com/single-spa/single-spa-angular/pull/398/files?file-filters%5B%5D=.json&show-deleted-files=true&show-viewed-files=true#diff-00f220e9170615038a66ec9ac1040e1ea5885b73373769b595304580b2d83c91).
2. Since we're building with Angular 13, has the latest Angular Package Format (APF) changes, primarily that UMD bundles aren't generated anymore. For example, on building you'll only see flattened ES2020 and ES2015 now:
```bash
------------------------------------------------------------------------------
Building entry point 'single-spa-angular'
------------------------------------------------------------------------------
✔ Compiling with Angular sources in Ivy partial compilation mode.
✔ Generating FESM2020
✔ Generating FESM2015
✔ Copying assets
✔ Writing package manifest
✔ Built single-spa-angular
```

Just wanted to say thank to you all for keeping Angular working with single-spa! I will try to contribute where I can, as where I work we're now building an Angular single-spa site, so hopefully I'll be able to have more insight / working experience with it.

**Comment by jonatanmartinbabel** [comment]
It seems that @joeldenning abandoned the project. I doubt he intends to approve anything else.
_Should someone start a new one (or a fork)?_

**Comment by joeldenning** [comment]
I took a break from open source in late November through December. I have come back to it in January and am sifting through a few hundred notifications little by little. I have not prioritized this PR specifically because I've had a half dozen people DM me about it, some of whom have been pretty rude to me and consistently violated my boundaries by repeatedly DMing me in multiple places despite me asking them not to. I have gotten paid a total of less than $3,000 for 6 years of work on single-spa, and have burned myself out several times in the process. So now I am slowly reintroducing myself to open source in a way I hope to be more sustainable.

I will get to this PR when I get to it, and hopefully be able to merge and publish a new version without too many rounds of PR feedback. Repeated mentioning or DMing of me will delay this happening, since I don't like when people try to get me to do things and intentionally deprioritize the things they want me to do.

For three years or so, I've hoped to find maintainers who can take over this project, since I personally do not enjoy Angular and find it to be the most difficult framework to support, largely due to Angular CLI and Ivy being incompatible with a lot of single-spa's objectives. Arturo is the best maintainer the project has, but he doesn't generally merge pull requests or publish versions. If you or your company is invested in this project, consider watching the GitHub repo and helping respond to issues and pull requests. I give out write and publish access pretty quickly once I see someone who's consistently helping out the project.

Below is a link to a blog post I wrote about how I intentionally avoid doing work that people violate my boundaries try to get me to do. Before DMing me to review this pull request, please read this blog post and then do not DM me.

https://joeldenning.com/?p=29

**Comment by chucknelson** [comment]
Appreciate the transparency @joeldenning, and it's pretty sad to hear some of those stories. People confusing open source with paid product support or something 😩 

My company is moving forward with single-spa and Angular, so hopefully we can turn that into a way to make time to contribute to this and the other single-spa projects. 🚀

File: libs/single-spa-angular/package.json
Comment (joeldenning) [inline libs/single-spa-angular/package.json:30]: This doesn't look like a good thing to do - new versions of npm autoinstall peer dependencies rather than using them, so this might not actually end up with the intended behavior. Instead of using Angular CLI's version of style-loader, it could be a different one.

File: libs/single-spa-angular/package.json
Comment (joeldenning) [inline libs/single-spa-angular/package.json:27]: This change likely requires that this pull request be published as a new breaking change in single-spa-angular@6 (which is okay)

File: libs/single-spa-angular/internals/package.json
Comment (joeldenning) [inline libs/single-spa-angular/internals/package.json:6]: This was removed, does this mean that single-spa-angular will no longer produce umd bundles? If so, that may break things for people relying on it loading as an in-browser systemjs module

File: libs/single-spa-angular/webpack/index.ts
Comment (joeldenning) [inline libs/single-spa-angular/webpack/index.ts:52]: The empty object seems like it could be removed, but already was in the code so i guess it's fine

File: package.json
Comment (joeldenning) [inline package.json:81]: Changes to these devDependencies also probably will result in build artifacts that don't work with Angular 12 - another reason to publish this as single-spa-angular@6

File: package.json
Comment (joeldenning) [inline package.json:116]: 👍🏿  thanks for properly upgrading from husky 4 by creating the .husky directory with commit-msg, etc

File: schematics/ng-add/rules/add-dependencies.ts
Comment (joeldenning) [inline schematics/ng-add/rules/add-dependencies.ts:66]: This makes me a bit nervous because ideally we'd use the version of style-loader included in angular devkit or angular cli, rather than our own peer dependency. Was style-loader removed from those? Is there a way we can use it?

**Review by joeldenning — APPROVED** [review]
I understood about 40% of the code changes here, and didn't spend much time testing or reviewing it. So I don't know if this works. I've left a few comments that normally would prevent me from approving and merging, but since in this case it has been such a long time since the pull request was opened and there are so many people wanting it published, I'm just going to ignore those comments and merge and publish it. If it results in anything not working, please create pull requests to fix.

**Comment by joeldenning** [comment]
Published in https://github.com/single-spa/single-spa-angular/releases/tag/v6.0.0. Like I mentioned above, I wasn't sure about a couple of the changes and didn't understand much of this, so if there are issues please send in pull requests to fix.

**Comment by joeldenning** [comment]
Documented at https://single-spa.js.org/docs/ecosystem-angular#angular-13

File: libs/single-spa-angular/package.json
Comment (jogelin) [reply libs/single-spa-angular/package.json:30]: @arturovt I think this is a valid comment, what is the reason behind this new dependency?

File: libs/single-spa-angular/internals/package.json
Comment (jogelin) [reply libs/single-spa-angular/internals/package.json:6]: @joeldenning Indeed, Angular 13 makes the format by default the usage of ESM as described here: [Angular Package Format](https://angular.io/guide/angular-package-format). But I am not an expert in bundles/webpack so I don't know what is the impact on single-spa?!

I know there are some repositories for allowing angular to be imported using systemJs (https://github.com/esm-bundle/angular__core) but is this format not compatible with systemJs?

@arturovt @chucknelson I think we should have a look at this question.

Comment (jogelin) [reply libs/single-spa-angular/internals/package.json:6]: after a quick, this is indeed bad:
- https://github.com/evanw/esbuild/issues/192
- https://github.com/evanw/esbuild/pull/255
- https://github.com/evanw/esbuild/issues/507

File: libs/single-spa-angular/package.json
Comment (arturovt) [reply libs/single-spa-angular/package.json:30]: https://github.com/angular/angular-cli/pull/21470/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L214

Comment (jogelin) [reply libs/single-spa-angular/package.json:30]: Ah ok, Angular Cli removed it then, because we need it for injecting CSS when loading the micro-frontend, you add it again.

Thx for the precisions @arturovt

File: libs/single-spa-angular/internals/package.json
Comment (chucknelson) [reply libs/single-spa-angular/internals/package.json:6]: @jogelin - I think even if `esbuild` supported outputting in UMD bundles, `ng-packagr` would still need to support outputting libraries in that as well. To my knowledge there are no plans to customize build output formats or anything, though, since `ng-packagr` is all about conforming to whatever the latest Angular Package Format (APF) spec is.

Maybe there are some clever workarounds, post-processing or something to get Angular 13+ built libraries to work as in-browser modules, but I'm not sure - nor do I know if that'll seem "right" in any way, might feel too hacky?

My experience with shared Angular has been spotty already (see https://github.com/single-spa/single-spa.js.org/issues/490), so maybe I'm missing something. @arturovt might know more.

From my experience thus far, tree-shaking in Angular works pretty well for keeping MFE bundle sizes small, so my teams have used "plain" Typescript/Javascript libraries for in-browser loading, which are utilized by native Angular libraries or apps that use them via webpack externals.

**Pull request merged**
